core/state.py

```diff
@@ -1497,6 +1497,31 @@ def GetAllVars(self):
           result[name] = str_val.s
     return result
 
+  def GetAllCells(self, lookup_mode = scope_e.Dynamic):
+    # type: () -> Dict[str, cell]
+    """Get all variables and their values, for 'set' builtin. """
+    result = {}  # type: Dict[str, str]
+
+    if lookup_mode == scope_e.Dynamic:
+      scopes = self.var_stack
+    elif lookup_mode == scope_e.LocalOnly:
+      scopes = self.var_stack[-1:]
+    elif lookup_mode == scope_e.GlobalOnly:
+      scopes = self.var_stack[0:1]
+    elif lookup_mode == scope_e.LocalOrGlobal:
+      scopes = self.var_stack[0:1]
+      if len(self.var_stack) > 1:
+        scopes.append(self.var_stack[-1])
+    else:
+      raise AssertionError()
+
+    for scope in scopes:
+      for name, cell in iteritems(scope):
+        result[name] = cell
+    return result
+
+  def IsGlobalScope(self):
```

Comment:
this should have `# type: () -> bool`

TODO: I think we can set up the continuous build to alert of that, since type checking is running there now

Reply:
Thank you! Updated 9e94f900.